model/course.py

```diff
@@ -97,3 +97,26 @@ def update_students(user, course, users, action):
             },
         )
     return HTTPResponse('success')
+
+
+@course_api.route('/<name>/tag', methods=['PATCH'])
+@login_required
+@Request.json('push: list', 'pop: list')
+@Request.doc('name', 'course', Course)
+@identity_verify(0, 1)  # only admin and teacher can call this route
+def update_tags(course, push, pop):
+    '''
+    push/pop tags to/from course
+    '''
+    for t in push:
+        if t not in engine.Tag.objects:
+            return HTTPError('Push: Tag not found', 404)
+        if t in pop:
+            return HTTPError('Tag appears in both list', 400)
+    for t in pop:
+        if t not in Course.tags:
+            return HTTPError('Pop: Tag not found', 404)
```

Comment:
compare string with a document, maybe you can use `if Tag(t)` to check document existence.